### ISSUE TITLE
reference of vg desconstruct must be one contig?

### ISSUE THREAD

**Issue by zhanghaoipp** [vgteam/vg#3062]
I have 100+ fungi assemblies, they were aligned by cactus and convert to vg graph. I try to use vg desconstruct to cal variants. But I can use strain name as reference, it just work when I select one contig. How to set strains names with several contigs as reference in this condition? Thank you!

**Comment by glennhickey** [comment]
Does using `-p contig1 -p contig2 -p contig3 etc.` not work?

**Comment by zhanghaoipp** [comment]
> Does using `-p contig1 -p contig2 -p contig3 etc.` not work?

Thank you Glennhickey! It works! But it is still different from normal vcf, the column is contigs but not isolates, maybe contigs from same isolate are can be merged?

The readme showed the basic usage is mapping and call. I also have raw reads of all strains, so can I map reads to the cactus graph individually and use vg call to call variants then merge all the vcfs? Is there any difference with desconstruct? Which is recommended? Thank you!

**Comment by glennhickey** [comment]
Do you mean the first two VCF columns look incorrect?  I'm not sure there's
much that can be done if so, as each contig (column 1) is a separate path
on your graph and has an independent coordinate system (column 2).

If you mean that there are too many sample columns (column 10 and beyond),
then yes the default output which creates a sample for each path in the
graph isn't going to help.  For now the only way to merge these is with the
-A option, and that relies on the contigs of each isolate having a unique
prefix in their name.  If you can enforce that, then the VCF you get should
work.  Otherwise, we'd need an option to specify which names to merge,
which would not be difficult to implement.  It's also important in general
to use -e, as well as to set the ploidy with -d.

I think deconstruct is a cleaner way to get a VCF than remapping, at least
in theory.  The map/call output should be generally consistent though.

**Comment by zhanghaoipp** [comment]
> sample columns

Yes, I mean the sample columns. The contig name like "140001.NODE_25_length_505019_cov_14.525173", 140001 is the prefix of the isolate, the later part is the contig name from spades. So I need to list all 100+ isolates (14002, 140003 ......) names seperate by comma behind -A?

**Comment by glennhickey** [comment]
Yeah, if you have 100 isolates, then you need to list 100 prefixes with -A
prefix1 -A prefix2 -A prefix3 etc.

**Comment by faithokamoto** [comment]
Seems this was answered at the time.
